sktime/annotation/hidalgo.py

```diff
@@ -0,0 +1,699 @@
+# -*- coding: utf-8 -*-
+
+"""
+HidAlgo (Heterogeneous Intrinsic Dimensionality Algorithm) Segmentation.
+
+Notes
+-----
+As described in
+@article{allegra2020data,
+  title={Data segmentation based on the local intrinsic dimension},
+  author={Allegra, Michele and Facco, Elena and Denti, Francesco and Laio,
+        Alessandro and Mira, Antonietta},
+  journal={Scientific reports},
+  volume={10},
+  number={1},
+  pages={1--12},
+  year={2020},
+  publisher={Nature Publishing Group}
+}
+"""
+
+from sktime.annotation.base import BaseSeriesAnnotator
+
+__author__ = ["KatieBuc"]
+__all__ = ["Hidalgo"]
+
+
+from functools import reduce
+
+import numpy as np
+from sklearn.neighbors import NearestNeighbors
+from sklearn.utils.validation import check_random_state
+
+
+def binom(N, q):
+    """Calculate the binomial coefficient."""
+    if q == 0:
+        return 1.0
+    if N < 0:
+        return 0.0
+    return reduce(lambda x, y: x * y, [(N - q1) / (q1 + 1) for q1 in range(q)])
+
+
+def Zpart(N, N1, zeta, q):
+    """Partition function for Z."""
+    return sum(
+        [
+            binom(N1 - 1, q1)
+            * binom(N - N1, q - q1)
+            * zeta ** (q1)
+            * (1 - zeta) ** (q - q1)
+            for q1 in range(q + 1)
+        ]
+    )
+
+
+class Hidalgo(BaseSeriesAnnotator):
+    """Class to fit parameters of the HidAlgo intrinsic dimension model.
+
+    explain, reference
```

Comment:
yes would be nice to have a simple description of that this algo is actually doing, so that people like me can also understand it :)